pages/radio-group/progressive-disclosure.page.tsx

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useReducer, useState } from 'react';
+import Box from '~components/box';
+import FormField from '~components/form-field';
+import RadioGroup from '~components/radio-group';
+import Select, { SelectProps } from '~components/select';
+
+const langOptions: Array<SelectProps.Option> = [
+  { value: 'en-US', label: 'English (United States)' },
+  { value: 'en-GB', label: 'English (United Kingdom)' },
+  { value: 'en-NZ', label: 'English (New Zealand)' },
+  { value: 'de-DE', label: 'Deutsch (Deutschland)' },
+  { value: 'de-CH', label: 'Deutsch (Schweiz)' },
+  { value: 'de-AT', label: 'Deutsch (Österreich)' },
+];
+
+const optionsPerLanguage = langOptions.reduce((acc, option) => {
+  const [lang] = option.value!.split('-');
+  if (!acc[lang]) {
+    acc[lang] = [];
+  }
+  acc[lang].push(option);
+  return acc;
+}, {} as Record<string, Array<SelectProps.Option>>);
+
+const initialState = {
+  en: optionsPerLanguage.en[0],
+  de: optionsPerLanguage.de[0],
+};
+
+type State = { [K in keyof typeof initialState]: SelectProps.Option };
+interface Action {
+  type: `update/${string & keyof State}`;
+  payload: Partial<State[keyof State]>;
+}
+
+const langReducer: React.Reducer<State, Action> = (state = initialState, action) => {
+  switch (action.type) {
+    case 'update/en':
+      return { ...state, en: action.payload };
+    case 'update/de':
+      return { ...state, de: action.payload };
+    default:
+      return state;
+  }
+};
+
+export default function RadiosPage() {
```

Comment:
Impressive demo, but a little bit over-engineered for this case. It is extra complexity when somebody tries to read this demo and apply it for their case

1. Remove `useReducer`, plain state will do
2. Remove extra options from radio group, we only need to show case how to show/hide an extra control


Basically, this demo should show a minimal reproducible example of `ariaControls` use-case.